Archiver rescheduling task crashes with RejectedExecutionException when executor is already terminated

## Description
The Camunda exporter's archiver rescheduling task throws a `RejectedExecutionException` (wrapped in `CompletionException`) when it attempts to reschedule itself after the underlying `ScheduledThreadPoolExecutor` has already been terminated. This appears to happen during component shutdown and is likely noise, but the log message ("task will NOT be retried") may be alarming and the exception should be handled gracefully regardless.

## Current Behavior
When the Elasticsearch connection is closed (e.g. during a restart or network disruption), the in-flight async request fails. The failure propagates through `AsyncRepeatUntil`, which attempts to reschedule the next run via the executor — but the executor has already been shut down (`Terminated, pool size = 0`). This results in:

```
Unexpected error occurred while rescheduling task process-instance-by-id archiver job (bug in error handling code?); task will NOT be retried
```

Full exception:

```
java.util.concurrent.CompletionException: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@43d81b43[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@5f1c16bf[Wrapped task = java.util.concurrent.CompletableFuture$UniExceptionally@57923315]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@1906a7f5[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 3486]

	at java.base/java.util.concurrent.CompletableFuture.wrapInCompletionException(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source)
	at io.camunda.exporter.tasks.archiver.AsyncRepeatUntil.completeExceptionally(AsyncRepeatUntil.java:53)
	at java.base/java.util.concurrent.CompletableFuture.uniExceptionally(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniExceptionally.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source)
	at co.elastic.clients.transport.ElasticsearchTransportBase.lambda$performRequestAsync$0(ElasticsearchTransportBase.java:199)
	at java.base/java.util.concurrent.CompletableFuture.uniHandle(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniHandle.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source)
	at co.elastic.clients.transport.rest_client.RestClientHttpClient$1.onFailure(RestClientHttpClient.java:121)
	at org.elasticsearch.client.RestClient$FailureTrackingResponseListener.onDefinitiveFailure(RestClient.java:688)
	at org.elasticsearch.client.RestClient$1.failed(RestClient.java:426)
	at org.apache.http.concurrent.BasicFuture.failed(BasicFuture.java:137)
	at org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.executionFailed(DefaultClientExchangeHandlerImpl.java:101)
	at org.apache.http.impl.nio.client.AbstractClientExchangeHandler.failed(AbstractClientExchangeHandler.java:432)
	at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.closed(HttpAsyncRequestExecutor.java:146)
	at org.apache.http.impl.nio.client.InternalIODispatch.onClosed(InternalIODispatch.java:77)
	at org.apache.http.impl.nio.client.InternalIODispatch.onClosed(InternalIODispatch.java:40)
	at org.apache.http.impl.nio.reactor.AbstractIODispatch.disconnected(AbstractIODispatch.java:100)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionClosed(BaseIOReactor.java:277)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.processClosedSessions(AbstractIOReactor.java:449)
	at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:283)
	at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.util.concurrent.RejectedExecutionException: Task java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask@43d81b43[Not completed, task = java.util.concurrent.Executors$RunnableAdapter@5f1c16bf[Wrapped task = java.util.concurrent.CompletableFuture$UniExceptionally@57923315]] rejected from java.util.concurrent.ScheduledThreadPoolExecutor@1906a7f5[Terminated, pool size = 0, active threads = 0, queued tasks = 0, completed tasks = 3486]
	at java.base/java.util.concurrent.ThreadPoolExecutor$AbortPolicy.rejectedExecution(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.reject(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.delayedExecute(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.schedule(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor.execute(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniCompletion.claim(Unknown Source)
	... 30 more
```

## Expected Behavior
When the executor is terminated (e.g. because the component is shutting down), the `RejectedExecutionException` should be handled gracefully — silently swallowed if shutdown is in progress, or the rescheduling logic should guard against submitting to a terminated executor before calling `schedule()`.

## Environment

- SaaS & SM: Both
- Version: 8.10.0-SNAPSHOT
- Component: Zeebe (Camunda Exporter / Archiver)

--- a/zeebe/exporter-common/src/main/java/io/camunda/zeebe/exporter/common/tasks/ReschedulingTask.java
+++ b/zeebe/exporter-common/src/main/java/io/camunda/zeebe/exporter/common/tasks/ReschedulingTask.java
@@ -12,6 +12,7 @@ import io.camunda.zeebe.util.ExponentialBackoff;
 import io.camunda.zeebe.util.VisibleForTesting;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
+import java.util.concurrent.RejectedExecutionException;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicLong;
@@ -82,6 +83,12 @@ public final class ReschedulingTask implements RunnableTask {
         .thenAccept(unused -> executionCounter.incrementAndGet())
         .exceptionally(
             error -> {
+              final var cause = error.getCause() != null ? error.getCause() : error;
+              if (closed && cause instanceof RejectedExecutionException) {
+                logger.debug(
+                    "Task {} was rejected because executor is shutting down; not retrying", task);
+                return null;
+              }
               logger.error(
                   "Unexpected error occurred while rescheduling task {} (bug in error handling code?);"
                       + " task will NOT be retried",

--- a/zeebe/exporter-common/src/test/java/io/camunda/zeebe/exporter/common/tasks/ReschedulingTaskTest.java
+++ b/zeebe/exporter-common/src/test/java/io/camunda/zeebe/exporter/common/tasks/ReschedulingTaskTest.java
@@ -8,7 +8,12 @@
 package io.camunda.zeebe.exporter.common.tasks;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.timeout;
+import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
 import java.util.concurrent.CompletableFuture;
@@ -179,6 +184,26 @@ final class ReschedulingTaskTest {
     inOrder
         .verify(EXECUTOR, Mockito.timeout(5_000).times(1))
         .schedule(task, 12L, TimeUnit.MILLISECONDS);
+  }
+
+  @Test
+  void shouldLogDebugNotErrorWhenRejectedDuringShutdown() throws InterruptedException {
+    // given
+    final var future = new CompletableFuture<Integer>();
+    when(archiverJob.execute()).thenReturn(future);
+    final var mockLogger = mock(Logger.class);
+    final var localExecutor = new ScheduledThreadPoolExecutor(1);
+    final var task = new ReschedulingTask(archiverJob, 1, 10, 1000, localExecutor, mockLogger);
+
+    // when - task is started, closed, executor shut down, then the in-flight future completes
+    task.run();
+    task.close();
+    localExecutor.shutdown();
+    future.complete(1);
+
+    // then - DEBUG log for shutdown rejection, no ERROR log
+    verify(mockLogger, timeout(5_000).atLeastOnce()).debug(anyString(), any(Object.class));
+    verify(mockLogger, never()).error(anyString(), any(Object.class), any(Throwable.class));
   }
 
   @Test

--- a/zeebe/exporter-common/src/test/java/io/camunda/zeebe/exporter/common/tasks/ReschedulingTaskTest.java
+++ b/zeebe/exporter-common/src/test/java/io/camunda/zeebe/exporter/common/tasks/ReschedulingTaskTest.java
@@ -10,6 +10,7 @@ package io.camunda.zeebe.exporter.common.tasks;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.timeout;
@@ -202,7 +203,10 @@ final class ReschedulingTaskTest {
     future.complete(1);
 
     // then - DEBUG log for shutdown rejection, no ERROR log
-    verify(mockLogger, timeout(5_000).atLeastOnce()).debug(anyString(), any(Object.class));
+    verify(mockLogger, timeout(5_000).times(1))
+        .debug(
+            eq("Task {} was rejected because executor is shutting down; not retrying"),
+            any(Object.class));
     verify(mockLogger, never()).error(anyString(), any(Object.class), any(Throwable.class));
   }
 

--- a/zeebe/exporter-common/src/test/java/io/camunda/zeebe/exporter/common/tasks/ReschedulingTaskTest.java
+++ b/zeebe/exporter-common/src/test/java/io/camunda/zeebe/exporter/common/tasks/ReschedulingTaskTest.java
@@ -188,7 +188,7 @@ final class ReschedulingTaskTest {
   }
 
   @Test
-  void shouldLogDebugNotErrorWhenRejectedDuringShutdown() throws InterruptedException {
+  void shouldLogDebugNotErrorWhenRejectedDuringShutdown() {
     // given
     final var future = new CompletableFuture<Integer>();
     when(archiverJob.execute()).thenReturn(future);

